test/js/specs/happy.js

```diff
@@ -2,125 +2,424 @@ const path = require('path')
-    expect(identityCardHintText).to.equal(documentSelectionLocale["document_selector"]["identity"]["national_identity_card_hint"])
-    documentSelection.identityCardHint.isDisplayed()
+    it('should display identity card hint', async () => {
+      const identityCardHintText = documentSelection.identityCardHint.getText()
+      expect(identityCardHintText).to.equal(documentSelectionLocale["document_selector"]["identity"]["national_identity_card_hint"])
+      documentSelection.identityCardHint.isDisplayed()
+    })
   })
 })
 
-  describe('upload screen', function () {
-    it('should upload a file', async () => {
+  //Document upload screen
+  describe('document upload screen', function () {
+
+    const copy = locale("en")
+    const documentUploadLocale = copy
+    const documentUploadConfirmationLocale = copy
+    const goToPassportUploadScreen = async (parameter) => {
+      if (typeof parameter === 'undefined') {
+          parameter = ''
+        }
+      driver.get(localhostUrl + parameter)
+      welcome.primaryBtn.click()
+      documentSelection.passportIcon.click()
+  }
+
+    function waitForUploadToFinish() { return (async () => {
+      const confirmBtn = await this.$('.onfido-sdk-ui-Confirm-btn-primary')
+      await driver.wait(until.elementIsVisible(confirmBtn),5000);
+    })}
+
+    it('should display cross device icon', async () => {
+      goToPassportUploadScreen()
+      documentUpload.crossDeviceIcon.isDisplayed()
+    })
+
+    it('should display cross device header', async () => {
+      const crossDeviceHeaderText = documentUpload.crossDeviceHeader.getText()
+      expect(crossDeviceHeaderText).to.equal(documentUploadLocale["cross_device"]["switch_device"]["header"])
+      documentUpload.crossDeviceHeader.isDisplayed()
+    })
+
+    it('should display cross device submessage', async () => {
+      const crossDeviceSubMessageText = documentUpload.crossDeviceSubMessage.getText()
+      expect(crossDeviceSubMessageText).to.equal(documentUploadLocale["cross_device"]["switch_device"]["submessage"])
+      documentUpload.crossDeviceSubMessage.isDisplayed()
+    })
+
+    it('should display cross device arrow', async () => {
+      documentUpload.crossDeviceArrow.isDisplayed()
+    })
+
+    it('should display uploader icon', async () => {
+      documentUpload.uploaderIcon.isDisplayed()
+    })
+
+    it('should display uploader button', async () => {
+      const uploaderBtnText = documentUpload.uploaderBtn.getText()
+      expect(uploaderBtnText).to.equal(documentUploadLocale["capture"]["upload_file"])
+      documentUpload.uploaderBtn.isDisplayed()
+    })
+
+    it('should upload a passport and verify UI elements', async () => {
+      goToPassportUploadScreen()
+      const passportTitleText =  documentUpload.title.getText()
+      expect(passportTitleText).to.equal(documentUploadLocale["capture"]["passport"]["front"]["title"])
+      documentUpload.title.isDisplayed()
```

Comment:
I was thinking about it but kept that as the additional check. 

I'm not sure whether there might be a situation that the element is not shown but the text of it might be passed anyway hence I added an additional check.

What isDisplayed() method does is:

```Determines whether an element is what a user would call "shown". This means that the element is shown in the viewport of the browser, and only has height and width greater than 0px, and that its visibility is not "hidden" and its display property is not "none"```

@stefaniacardenas if you think the situation I described above cannot happen, I'm happy to remove it when unnecessary.